src/ui/public/chrome/services/breadcrumb_state.js

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { uiModules } from '../../modules';
+import { Observable } from 'rxjs';
+
+let observer;
+const observable = Observable.create((o) => observer = o);
+
+export const set = (breadcrumbs) => observer && observer.next(breadcrumbs);
+
+uiModules.get('kibana')
+  .service('breadcrumbState', ($rootScope) => {
+    $rootScope.$on('$routeChangeStart', () => set([]));
```

Comment:
So, this "works" to make sure breadcrumbs don't hang around between pages, but it does cause some flickering in some cases where it will go back to the default (eg. "Kibana") and then flip to the new breadcrumbs.

The only _good_ alternative I can think of is requiring that plugins clear / set the breadcrumbs explicitly on routes. Good thing is if the plugin doesn't use custom breadcrumbs at all, it will stay as the app name or if they use the `<bread-crumbs>` directive everything just works.